version: "1"
services:
  database:
    image: postgres:14.6-alpine
    restart: always
    environment:
      - POSTGRES_DB=teamifydb
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "sh -c 'pg_isready -U postgres -d teamifydb'"]
      interval: 10s
      timeout: 3s
      retries: 3
    # volumes:
    #   - ./database:/var/lib/postgresql/data

  server:
    build: ./server
    ports:
      - "8000:8000"
    depends_on:
      - database
    environment:
      POSTGRES_DB: teamifydb
      POSTGRES_HOST: database
      POSTGRES_PORT: 5432
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      DATABASE_URL: postgres://postgres:postgres@database:5432/teamifydb
    volumes:
      - ./server:/server
    command: ash -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"

  web:
    build: ./web
    environment:
      NEXT_PUBLIC_SERVER_URL: http://localhost:8000
    ports:
      - "3000:3000"
    depends_on:
      - server
    volumes:
      - ./web:/web
    command: "yarn && yarn dev"

volumes:
  db:
    driver: local
